src/AzureExtension/Services/DevBox/DevBoxCreationManager.cs

```diff
@@ -142,6 +142,14 @@ public bool TryGetDevBoxInstanceIfBeingCreated(string id, out DevBoxInstance? de
 
         lock (_creationLock)
         {
+            // The user previously chose to delete the Dev Box while it was being created. So it is no longer being
+            // watched and we need to remove it from the creation map.
+            if (!_devBoxOperationWatcher.IsIdBeingWatched(Guid.Parse(id)) && _devBoxesBeingCreated.ContainsKey(id))
```

Comment:
No need to check if id is in the dictionary. Remove will do it for you anyway.

Reply:
oh noted for next time.